### PR TITLE
refactor(robot-server): deck cal - jog to deck and calibration points

### LINKED ISSUE #6349: Deck Calibration: Implement jog to deck (robot side)
Implement the jog to deck actions required by the [jog to deck state](https://www.figma.com/file/sbjDtrRzqd46SEx1NG5QHD/Calibration-Overhaul-Deck-Calibration?node-id=46%3A6) in deck calibration.

- The state should allow z jog
- When the user transitions out of the state, the user flow should save the current z position for use when moving to the three calibration points

### LINKED ISSUE #6351: Deck Calibration: Implement jog-to-calibration point (robot side)
Implement the three jog-to-calibration-point states: [one](https://www.figma.com/file/sbjDtrRzqd46SEx1NG5QHD/Calibration-Overhaul-Deck-Calibration?node-id=46%3A7), [two](https://www.figma.com/file/sbjDtrRzqd46SEx1NG5QHD/Calibration-Overhaul-Deck-Calibration?node-id=46%3A8), [three](https://www.figma.com/file/sbjDtrRzqd46SEx1NG5QHD/Calibration-Overhaul-Deck-Calibration?node-id=46%3A9) in the user flow.

Each state should begin with the robot jogging to the nominal x-y position of the point, at the z-position saved in jog-to-deck.
Each state should allow x and y jog
Each state should save the corresponding point on exit

### PR DESCRIPTION
**Pull request by ahiuchingau**
closes #6349, closes #6351

# Overview
This PR implements the (1) jogging to deck, and (2) jogging to calibration-points in the deck calibration flow. 

# Changelog
- filled in `save_offset()`
  - during `joggingToDeck`, saves only the z as the deck z reference height to be used in determining the move to height for the 3 calibration points
  - during `savingPoint[One,Two,Three]`, adds the current position to `saved_points`
- adding `get_cal_point_location` to get the location for each calibration point, using the deck z reference height as the z

# Review requests
Note: This PR doesn't take care of the actual saving the attitude matrix yet! It will be done in a separate PR.

# Risk assessment
Low, behind a feature flag, all new code

### CODE REVIEW

File: robot-server/robot_server/robot/calibration/deck/user_flow.py [hunk 1]
```diff
@@ -55,6 +61,9 @@ def __init__(self,
         self._current_state = State.sessionStarted
         self._state_machine = DeckCalibrationStateMachine()
 
+        self._z_height_reference: Optional[float] = None
+        self._saved_points: tuple = ()
```
Comment (sfoster1) [inline robot-server/robot_server/robot/calibration/deck/user_flow.py:65]: this isn't the right way to type this. `Tuple`s in are fixed-length containers for heterogeneous data (or, to support some pythonisms, unlimited length "immutable" containers for homogenous data, but gross) and therefore need a different approach to typing and to consideration for reusing them in class members like this. We could and possibly should type this as `self._saved_points = Tuple[Optional[float], Optional[float], Optional[float]] = (None, None, None)`. We could also, and I think this is a better idea, make a typed dict like
```
class SavedPoints(TypedDict):
   '1BRC': float  # or whatever the idea is
   '3BRC': float
   '7BRC': float
   'z': float  # or some other id
```

and then have `self._saved_points: SavedPoints = {}` or something like this. we can later pass the xy points to solve with `[v for v in value if k != 'z' for k, v in self._saved_points.items()]`

**Review by sfoster1 — CHANGES_REQUESTED** [review]
I think we should change the way we save points to make the typing work a bit better

**Review by b-cooper — APPROVED** [review]
Code looks good to me 🏯

**Review by sfoster1 — APPROVED** [review]
LGTM! Nice!

**Pull request merged**